src/test/java/guitests/PersonEditGuiTest.java

```diff
@@ -125,36 +126,70 @@ public void cancelOperation_usingAccelerator() {
                 .withStreet("Chengdu Panda Street").withCity("Chengdu").withPostalCode("PANDA")
                 .withBirthday("01.01.1979").withGithubUsername("panda").withTags(td.colleagues, td.friends).build();
-        addPersonDialog.enterNewValues(pandaWong);
-        addPersonDialog.clickOk();
+        addPersonDialog.enterNewValues(pandaWong).clickOk();
 
-        personListPanel.clickOnListView(); // To ensure shortcut keys work properly.
-
-        personListPanel.use_LIST_GOTO_BOTTOM_SEQUENCE();
+        personListPanel.navigateToPerson(pandaWong);
         PersonCardHandle pandaWongCardHandle = personListPanel.getPersonCardHandle(pandaWong);
-        assertEquals(pandaWongCardHandle, pandaWong);
+        assertTrue(pandaWongCardHandle.isShowingGracePeriod("Adding"));
+        assertEquals(pandaWongCardHandle, pandaWong); //Ensure correct state before cancelling.
+
         personListPanel.use_PERSON_CHANGE_CANCEL_ACCELERATOR();
         assertNull(personListPanel.getPersonCardHandle(pandaWong));
+        assertFalse(pandaWongCardHandle.isShowingGracePeriod("Adding"));
+        assertEquals("Add Person [ Panda Wong ] was cancelled.", statusBar.getText());
+    }
+
+    @Test
+    public void editPerson_editDuringGracePeriod() {
+        Person newAlice = new PersonBuilder(td.alice.copy()).withFirstName("Alicia").withLastName("Brownstone")
+                .withStreet("Updated street").withCity("Singapore").withPostalCode("123123")
+                .withBirthday("01.01.1979").withGithubUsername("alicebrown123").withTags(td.colleagues, td.friends).build();
 
-        //Ensure cancel operation has no side effects after grace period.
+        //Get a reference to the card displaying Alice's details
+        PersonCardHandle alicePersonCard = personListPanel.selectCard(td.alice);
+
+        //Edit Alice to change to new values
+        EditPersonDialogHandle editPersonDialog = personListPanel.use_PERSON_EDIT_ACCELERATOR();
+        editPersonDialog.enterNewValues(newAlice).pressEnter();
+
+        //Ensure grace period is showing
+        assertTrue(alicePersonCard.isShowingGracePeriod("Editing"));
+
+        //Edit Alice again during pending state.
+        personListPanel.clickOnPerson(newAlice);
+        Person newerAlice = new PersonBuilder(newAlice.copy()).withFirstName("Felicia").withLastName("Yellowstone")
+                .withStreet("street updated").withCity("Malaysia").withPostalCode("321321")
+                .withBirthday("11.11.1979").withGithubUsername("yellowstone").withTags(td.colleagues).build();
+
+        editPersonDialog = personListPanel.use_PERSON_EDIT_ACCELERATOR();
+
+        //Ensure grace period is not counting down while editing person.
+        assertTrue(alicePersonCard.isGracePeriodFrozen());
+
+        editPersonDialog.enterNewValues(newerAlice).pressEnter();
+
+        //Ensure card is displaying Felicia before and after grace period.
+        assertEquals(alicePersonCard, newerAlice);
```

Comment:
How do you confirm if the grace period restarted from 0? Perhaps the grace period was never activated?